test/test_distributions_util.py

```diff
@@ -17,70 +15,8 @@
     cumsum,
     multinomial,
     vec_to_tril_matrix,
-    xlog1py,
-    xlogy
 )
 
-_zeros = partial(lax.full_like, fill_value=0)
-
-
-@pytest.mark.parametrize('x, y', [
-    (np.array([1]), np.array([1, 2, 3])),
-    (np.array([0]), np.array([0, 0])),
-    (np.array([[0.], [0.]]), np.array([1., 2.])),
-])
-@pytest.mark.parametrize('jit_fn', [False, True])
-def test_xlogy(x, y, jit_fn):
-    fn = xlogy if not jit_fn else jit(xlogy)
-    assert_allclose(fn(x, y), osp_special.xlogy(x, y))
-
-
-@pytest.mark.parametrize('x, y, grad1, grad2', [
-    (np.array([1., 1., 1.]), np.array([1., 2., 3.]),
-     np.log(np.array([1, 2, 3])), np.array([1., 0.5, 1./3])),
-    (np.array([1.]), np.array([1., 2., 3.]),
-     np.sum(np.log(np.array([1, 2, 3]))), np.array([1., 0.5, 1./3])),
-    (np.array([1., 2., 3.]), np.array([2.]),
-     np.log(np.array([2., 2., 2.])), np.array([3.])),
-    (np.array([0.]), np.array([0., 0.]),
-     np.array([-float('inf')]), np.array([0., 0.])),
```

Comment:
Using upstream `xlogy` only makes the test for `grad2` here fail (the upstream version returns `nan`, which is caused by `0 / 0`). I think it should be fine to relax that semantic.